pentaho-js/src/main/javascript/cdf/components/XactionComponent.js

```diff
@@ -49,7 +50,10 @@ define([
                                   " height=\"100%\"" +
                                   " width=\"100%\" />";
           var iframe = $(xactionIFrameHTML);
-          var url = XactionComponentExt.getCdfXaction(this.path, this.action, this.solution) + "&wrapper=false";
+          var url = XactionComponentExt.getCdfXaction(myself.path,
+              Utils.pathIncludesAction(this.path, this.action) ? "" : this.action || "",
```

Comment:
I don't think we should need to know the precedence order to such detail, by heart. Unusual cases should just use parenteses, imo.

Please, change the last bit to `(this.action || "")` (assuming I guessed the correct precedence order).

Also, if a variable was defined to hold the effective `action` value, it would be much easier to read.

Reply:
I agree with your suggestion about parentheses ;)